libraries/package-deps-hash/src/getRepoState.ts

```diff
@@ -270,6 +277,149 @@ export function applyWorkingTreeState(
   }
 }
 
+/**
+ * Helper function for async process invocation with optional stdin support.
+ * @param gitPath - Path to the Git executable
+ * @param args - The process arguments
+ * @param currentWorkingDirectory - The working directory. Should be the repository root.
+ * @param stdin - An optional Readable stream to use as stdin to the process.
+ */
+async function spawnGitAsync(
+  gitPath: string | undefined,
+  args: string[],
+  currentWorkingDirectory: string,
+  stdin?: Readable
+): Promise<string> {
+  const spawnOptions: IExecutableSpawnOptions = {
+    currentWorkingDirectory,
+    stdio: ['pipe', 'pipe', 'pipe']
+  };
+
+  let stdout: string = '';
+  let stderr: string = '';
+
+  const proc: child_process.ChildProcess = Executable.spawn(gitPath || 'git', args, spawnOptions);
+  proc.stdout!.setEncoding('utf-8');
+  proc.stderr!.setEncoding('utf-8');
+
+  proc.stdout!.on('data', (chunk: string) => {
+    stdout += chunk.toString();
```

Comment:
Would it be more performant to append these chunks to an array and then `.join('')` then at the end?

Reply:
It's a memory vs. CPU trade-off. In V8 string concatenation by default just allocates a "cons string" object which is a head/tail pointer pair, so all computation of the completed string is deferred until the end.
Ultimately the performance impact of this part of the process is negligible; all our time is spent in the `git` process itself.